.github/workflows/check.yml

```diff
@@ -0,0 +1,70 @@
+# This workflow will install Python dependencies, run tests and lint both for the action server and core
+
+name: Python Checks
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  check-core:
+    runs-on: ubuntu-latest
+    env:
+      working-directory: ./core
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.6.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6.8
+      - name: Installing poetry
+        run: |
+          export $( \
+              PYTHONFAULTHANDLER=TRUE \
+              PYTHONUNBUFFERED=TRUE \
+              PIP_DISABLE_PIP_VERSION_CHECK=on \
+              PIP_NO_CACHE_DIR=off \
+          )
+          pip install --upgrade pip \
+          && pip install "poetry==1.0.5"
+      - name: Installing dependencies
+        run: make install
+        working-directory: ${{env.working-directory}}
+      - name: Linting
+        run: make lint
+        working-directory: ${{env.working-directory}}
+      - name: Unit testing
+        run: make test
+        working-directory: ${{env.working-directory}}
+
+  check-action-server:
```

Comment:
It's a new feature and I haven't had the chance to play with it yet, but it should be possible to streamline the pipeline and abstract the common logic in both jobs by using [Composite Run Steps](https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/).

However, I understand if you prefer not to take the time to investigate this right now. I don't think it's critical.

Reply:
I did play with it a little, and in this case it is not so useful because:
* A composite run steps action cannot call another one, so we still have to repeat the python setup step
* The jobs' defaults are not used, and we can't specify defaults for the composite action, meaning we have to specify the wd for each `make` step, using an input to the action. Add that to the fact that the `shell` field is required for each step there, those 3 are more cumbersome in the action than repeated.

So the only step that is simple to put in the action is the poetry step, but for one step, even though the bash command is ugly, I don't think it is worth separating the code.

Reply:
Changed my mind. It's cuter with a "setup-poetry" composite action.

Reply:
Alright, good to know those limitations. It seems there is some work in progress to address those: https://github.com/actions/runner/issues/646, I'll keep an eye on this.  

Thanks for taking the time to try it out.